docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+To generate a new `item_id`, let’s use the `incr` method from redis, passing a new hash "item_ids". When this method is run for the first time, it will create the item_ids hash with a unique number, and then each time it's run it will generate an incremental number and store it in this hash.
+
+8. Delete the line with the following content:
+    ```python
+    item_id: int = max(grocery_list.keys()) + 1 if grocery_list else 0
+    ```
+    And add the following:
+    ```python
+    item_id: int = redis_client.incr("item_ids")
+    ```
+
+Now we will add the item to the redis hash, using the `hset` method and by providing a mapping of fields or keys (`item_id`, `quantity` and `name`), and the values (the item’s newly created id, and its provided name and quantity).
+
+9.	Delete the line with the following content:
+    ```python
+    grocery_list[item_id] = ItemPayload(
+            item_id=item_id, item_name=item_name, quantity=quantity
+        )
+    ```
+    And replace it with the following:
+    ```python
+    redis_client.hset(
+                f"item_id:{item_id}",
+                mapping={
+                    "item_id": item_id,
+                    "item_name": item_name,
+                    "quantity": quantity,
+                })
+    ```
+
+Now we only need to map the newly created ID to the item name by setting the hash we referenced in the beginning, `item_name_to_id`.
+
+10.	Add this line to the end of the route, inside the `else` block:
+    ```python
+    redis_client.hset("item_name_to_id", item_name, item_id)
+    ```
+
+11. Delete the line with the following content:
+    ```python
+    return {"item": grocery_list[item_id]}
+    ```
+    And replace it with:
+    ```python
+    return {"item": ItemPayload(item_id=item_id, item_name=item_name, quantity=quantity)}
+    ```
+
+12.	If you would like, you can try to do a similar replacement for the other routes. Otherwise, you can just replace the entire content of the file with the lines below:
+    ```python
+    @app.get("/")
+    def home(request: Request) -> dict[str, str]:
+        url = str(request.base_url)
+        return {"message": f"Add /docs to the end of the URL to access the Swagger UI."}
+
+    # Route to add an item
+    @app.post("/items/{item_name}/{quantity}")
+    def add_item(item_name: str, quantity: int) -> dict[str, ItemPayload]:
+    if quantity <= 0:
+        raise HTTPException(status_code=400, detail="Quantity must be greater than 0.")
+
+    # Check if item already exists
+    item_id_str: str | None = redis_client.hget("item_name_to_id", item_name)
+
+    if item_id_str is not None:
+        item_id = int(item_id_str)
+        redis_client.hincrby(f"item_id:{item_id}", "quantity", quantity)
+    else:
+        # Generate an id for the item
+        item_id: int = redis_client.incr("item_id")
```

Comment:
In step 8 above, we added `redis_client.incr("item_ids")` here. Note that the two hash names are different. Is one of them wrong, or does it not matter?